### PR TITLE
Add module documentation for the server

### PR DESCRIPTION
**Pull request by jparise**
This adds some high-level documentation to the `Thrift` module and
detailed documentation to `Thrift.Binary.Framed.Server`.

### CODE REVIEW

File: lib/thrift/binary/framed/server.ex [hunk 1]
```diff
@@ -8,10 +14,45 @@ defmodule Thrift.Binary.Framed.Server do
...
+
+  `ssl_opts`: A keyword list of SSL/TLS options:
+
+  - `:enabled`: A boolean indicating whether to upgrade the connection to
+    the SSL protocol
+  - `:optional`: A boolean indicating whether to accept both SSL and plain
+    connections
+  - `:configure`: A 0-arity function to provide additional SSL options at
+    runtime
+  - Additional `t:Thrift.Transport.SSL.option/0` values specifying other
```
Comment (fishcakez) [inline lib/thrift/binary/framed/server.ex]: The additional options are `:ssl.ssl_option`?

File: lib/thrift.ex [hunk 2]
```diff
@@ -140,6 +140,42 @@ defmodule Thrift do
   [naming], so `getUser` becomes `get_user`.
 
   [naming]: http://elixir-lang.org/docs/stable/elixir/naming-conventions.html
+
+  ## Servers
+
+  Thrift servers are a little more involved because you need to create a module
+  to handle the work. Fortunately, a `Behaviour` is generated for each server
+  (complete with typespecs). Just use the `@behavior` module attribute, and the
```
Comment (fishcakez) [inline lib/thrift.ex]: s/Just use/Use/

File: lib/thrift/binary/framed/server.ex [hunk 3]
```diff
@@ -8,10 +14,45 @@ defmodule Thrift.Binary.Framed.Server do
...
 
   @type server_opts :: [server_option]
 
-  @spec start_link(module, 1..65_535, module, [server_option]) :: GenServer.on_start()
+  @doc """
+  Starts the server using the specified handler module.
+
+  The following server options can be specified:
+
+  `:worker_count`: The number of acceptor workers used to take requests.
```
Comment (fishcakez) [inline lib/thrift/binary/framed/server.ex]: acceptor workers to accept on the socket, different processes are used to handle requests

File: lib/thrift/binary/framed/server.ex [hunk 4]
```diff
@@ -8,10 +14,45 @@ defmodule Thrift.Binary.Framed.Server do
...
+
+  `:name`: (Optional) The name of the server. The server's pid becomes
+  registered under this name. If not specified, the handler module's name
+  is used.
+
+  `:max_restarts`: The number of times to restart (see `:max_seconds`).
```
Comment (fishcakez) [inline lib/thrift/binary/framed/server.ex]: I don't think we need to expose these. Instead set `max_restarts: 0` and allow the parent supervisor to handle things. The supervisor this configures is hidden from the user.

Comment (jparise) [reply lib/thrift/binary/framed/server.ex]: That sounds like more than a documentation, yes? I'll remove these from the documentation here, and we can do the other bits separately.

Comment (jparise) [reply lib/thrift/binary/framed/server.ex]: Done in #400.

**Pull request merged**